msCompile problem matcher does not match "fatal error"

Type: <b>Bug</b>

Including a non-existent file in C++ causes an error that does not get matched.
`C:\Users\akbyrd\Desktop\vscode-tasks\main.cpp(1): fatal error C1083: Cannot open include file: 'foo.h': No such file or directory`

https://github.com/microsoft/vscode/blob/fabb5745425b3e1836000995deea37601e3024a8/src/vs/workbench/contrib/tasks/common/problemMatcher.ts#L1287

The problem is that the regex will match "error" but not "fatal error" (or "fatal  error" with 2 spaces in the case of invoking msbuild instead of cl).

VS Code version: Code - Insiders 1.77.0-insider (c83f54aefc2dd212c01a42c57c930aa8a13af3a6, 2023-03-24T05:23:09.316Z)
OS version: Windows_NT x64 10.0.19045
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 9 5950X 16-Core Processor             (32 x 3400)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|31.92GB (21.05GB free)|
|Process Argv|--crash-reporter-id 22cf4e37-a96e-4ad9-82f0-8715f8e2bf6e|
|Screen Reader|no|
|VM|0%|
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30422396
pythontb:30258533
pythonptprofiler:30281269
vsdfh931cf:30280410
vshan820:30294714
pythondataviewer:30285072
vscod805cf:30301675
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30404738
cppdebug:30492333
vsclangdf:30492506
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30618038
pylantcb52:30590116
pyindex848:30611229
nodejswelcome1:30587009
pyind779:30611226
pythonsymbol12:30651887
2i9eh265:30646982
showlangstatbar:30659908
pythonb192cf:30661257
defaultazurewalk:30687958
pythonms35:30671666
j16hj152:30680472
ecj1e332:30687743

```

</details>

<!-- generated by issue reporter -->

I don't get a useful message about the problem in the problems view:

<img width="317" alt="image" src="https://github.com/microsoft/vscode/assets/38270282/3b652787-7772-4a4d-8a08-28d84a314618">

Test task (for Windows at least):
```
		{
			"type": "shell",
			"label": "test",
			"command": "echo 'C:\\Users\\akbyrd\\Desktop\\vscode-tasks\\main.cpp(6): fatal error C1083: Cannot open include file: ''foo.h'': No such file or directory'",
			"problemMatcher": "$msCompile"
		},
```

--- a/src/vs/workbench/contrib/tasks/common/problemMatcher.ts
+++ b/src/vs/workbench/contrib/tasks/common/problemMatcher.ts
@@ -1284,7 +1284,7 @@ class ProblemPatternRegistryImpl implements IProblemPatternRegistry {
 
 	private fillDefaults(): void {
 		this.add('msCompile', {
-			regexp: /^(?:\s*\d+>)?(\S.*)\((\d+|\d+,\d+|\d+,\d+,\d+,\d+)\)\s*:\s+((fatal +)?error|warning|info)\s+(\w+\d+)\s*:\s*(.*)$/,
+			regexp: /^(?:\s*\d+>)?(\S.*)\((\d+|\d+,\d+|\d+,\d+,\d+,\d+)\)\s*:\s+((?:fatal +)?error|warning|info)\s+(\w+\d+)\s*:\s*(.*)$/,
 			kind: ProblemLocationKind.Location,
 			file: 1,
 			location: 2,

